app/src/main/java/com/example/criminalintentapp/CrimeListFragment.kt

```diff
@@ -1,32 +1,57 @@
 package com.example.criminalintentapp
 
+import android.content.Context
 import android.os.Bundle
 import android.util.Log
 import android.view.View
+import android.widget.TextView
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
+import java.util.UUID
 
 class CrimeListFragment : Fragment(R.layout.fragment_crime_list) {
 
+    interface Callbacks {
+        fun onCrimeSelected(crimeId: UUID)
+    }
+
+    var callbacks: Callbacks? = null
+
     private lateinit var crimeRecyclerView: RecyclerView
+    private lateinit var emptyListTextView: TextView
     private var adapter: CrimeAdapter = CrimeAdapter(emptyList())
 
     private val crimeListViewModel: CrimeListViewModel by lazy {
         ViewModelProvider(this).get(CrimeListViewModel::class.java)
     }
 
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        callbacks = context as Callbacks?
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        emptyListTextView = view.findViewById(R.id.empty_list_text_view) as TextView
```

Comment:
Moving it here works but now the `setupUI()` is a bit misleading since it does not create this textview. I'd suggest to break the `setupUI()` method into two separate methods. The first will inflate all the views and can be called here, and the second will update the `crimes` list in the adapter. This will help solve another thing, when `crimeListViewModel.crimesListLiveData.observe` emits a new list of `crimes`there's no need to inflate all the views in `setupUI` again, just the `crimes` list needs to be updated in the adapter